pyxem/signals/diffraction2d.py

```diff
@@ -796,7 +813,28 @@ def find_peaks(self, method, *args, **kwargs):
                 "implementations.".format(method)
             )
 
-        peaks = self.map(method, *args, **kwargs, inplace=False, ragged=True)
+        if self._lazy:
+            dask_array = self.data
+        else:
+            sig_chunks = list(self.axes_manager.signal_shape)[::-1]
+            chunks = [8] * len(self.axes_manager.navigation_shape)
+            chunks.extend(sig_chunks)
+            dask_array = da.from_array(self.data, chunks=chunks)
+
+        output_array = dt._find_peak_dask_array(
+                dask_array,
+                method,
+                args_find_peak=args,
+                kwargs_find_peak=kwargs)
+
+        if show_progressbar:
+            pbar = ProgressBar()
+            pbar.register()
+        output_array_computed = output_array.compute()
+        if show_progressbar:
+            pbar.unregister()
+        peaks = hs.signals.BaseSignal(output_array_computed).T
+
         peaks.map(
             peaks_as_gvectors,
             center=np.array(self.axes_manager.signal_shape) / 2 - 0.5,
```

Comment:
This assumes the `Diffraction2D` object (self) has already been calibrated, with having the centre beam in the middle with the correct offsets.

Would it be ok to rather use `[self.axes_manager.signal_axes[0].value2index(0.0), self.axes_manager.signal_axes[1].value2index(0.0)]` instead? This would (I think) allow for both calibrated and non-calibrated datasets to work. Currently non-calibrated datasets end up with the `peaks` being in the top left corner.

For some big datasets, pre-calibrating is tricky. Working with the `peaks` themselves is much easier, since those are typically in the order of a couple 100 MB.

Reply:
For various reasons, we wanted to push people to perform specific calibration experiments so that this wasn't really an issue. I.e. they should have the calibration at least approximately before they need to do peak finding.

I think that a DiffractionVectors object does not make much sense without calibration but for various other reasons we intend to separate DiffractionVectors and DetectorCoordinates anyway see #485 

The PR #485 is a bit of a mess at this point so probably isn't very helpful - but I think we should take this opportunity to do this separation and create a DetectorCoordinates class - then I think find_peaks should return DetectorCoordinates if not calibrated and DiffractionVectors if calibrated.

Does that sound ok?